solidity/dashboard/src/sagas/coverage-pool.js

```diff
@@ -352,28 +325,71 @@ export function* subscribeToWithdrawalCompletedEvent() {
       payload: { event },
     } = yield take(requestChan)
     const {
-      returnValues: { underwriter, amount },
+      returnValues: { underwriter, amount, covAmount },
     } = event
     const address = yield select(selectors.getUserAddress)
 
-    if (isSameEthAddress(address, underwriter)) {
+    const { covTotalSupply, covBalance } = yield select(
+      selectors.getCoveragePool
+    )
+    const updatedCovTotalSupply = sub(covTotalSupply, amount)
+
+    if (!isSameEthAddress(address, underwriter)) {
       yield put(
-        showModal({
-          modalComponentType: modalComponentType.COV_POOLS.WITHDRAWAL_COMPLETED,
-          componentProps: {
-            transactionHash: event.transactionHash,
-            transactionFinished: true,
-            amount: amount,
-          },
-          modalProps: {
-            title: "Claim tokens",
-            classes: {
-              modalWrapperClassName: "modal-wrapper__claim-tokens",
-            },
-          },
+        covTokenUpdated({
+          covTotalSupply: updatedCovTotalSupply,
         })
       )
+      continue
     }
+
+    yield put(
+      showModal({
+        modalComponentType: modalComponentType.COV_POOLS.WITHDRAWAL_COMPLETED,
+        componentProps: {
+          transactionHash: event.transactionHash,
+          transactionFinished: true,
+          amount: amount,
+        },
+        modalProps: {
+          title: "Claim tokens",
+          classes: {
+            modalWrapperClassName: "modal-wrapper__claim-tokens",
+          },
+        },
+      })
+    )
+
+    const updatedCovBalance = sub(covBalance, covAmount)
+
+    const shareOfPool = yield call(
+      Keep.coveragePoolV1.shareOfPool,
+      updatedCovTotalSupply,
+      updatedCovBalance
+    )
+
+    const estimatedKeepBalance = yield call(
+      Keep.coveragePoolV1.estimatedCollateralTokenBalance,
+      shareOfPool
+    )
+
+    const estimatedRewards = yield call(
+      Keep.coveragePoolV1.estimatedRewards,
+      address,
+      shareOfPool
+    )
+
+    yield put(
+      covTokenUpdated({
+        pendingWithdrawal: "0",
+        withdrawalInitiatedTimestamp: "0",
+        shareOfPool,
+        covBalance: updatedCovBalance,
+        covTotalSupply: updatedCovTotalSupply,
+        estimatedRewards,
+        estimatedKeepBalance,
+      })
+    )
```

Comment:
We should also update `tvl` here because without that it will take the old value for calculating `covKeep -> KEEP` `and thus calculating it wrong.

Reply:
I think the solution for that would be to subtract `amount` from `tvl`

Reply:
Please see da1d84e